src/arithmetic/analyzer.cc

```diff
@@ -11,14 +11,18 @@ namespace arith {
 Analyzer::Analyzer()
     : const_int_bound(this),
       modular_set(this),
-      rewrite_simplify(this) {
+      rewrite_simplify(this),
+      canonical_simplify(this) {
 }
 
 void Analyzer::Bind(const VarExpr& v, const Expr& expr) {
   Var var(v.node_);
-  this->const_int_bound.Update(var, this->const_int_bound(expr));
-  this->modular_set.Update(var, this->modular_set(expr));
-  this->rewrite_simplify.Update(var, this->rewrite_simplify(expr));
+  Expr new_expr = expr;
```

Comment:
I don't understand why do you copy expr to `new_expr` here.

Reply:
To make new expr mutable

Reply:
But none of the subsequent lines changes it, or am I wrong?.